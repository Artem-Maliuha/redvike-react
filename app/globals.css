@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --primary: #1A1A1E;
  --background-rgb: 255, 255, 255;
}

body {
  color: rgb(var(--primary));
  background: rgb(var(--background-rgb));
}
