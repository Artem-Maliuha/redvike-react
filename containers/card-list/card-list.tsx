import { ReactNode } from 'react'

interface CardListProps {
  children: ReactNode[]
}
export const CardList = (props: CardListProps) => {
  const { children } = props

  return (
    <ul className="flex flex-col flex-wrap justify-between gap-y-[8px] sm:flex-row sm:gap-x-[16px] sm:gap-y-[16px]">
      {children.map((item, index) => (
        <li key={index} className="flex">
          {item}
        </li>
      ))}
    </ul>
  )
}
