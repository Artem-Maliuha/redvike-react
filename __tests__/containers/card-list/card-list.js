import renderer from 'react-test-renderer'
import { describe, it, expect } from '@jest/globals'
import { CardList } from '../../../containers/card-list'

describe('<CardList />', () => {
  it('snapshot', () => {
    const component = renderer.create(
      <CardList>
        {[...Array(10).keys()].map((value, index) => (
          <div key={index}>{value}</div>
        ))}
      </CardList>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
