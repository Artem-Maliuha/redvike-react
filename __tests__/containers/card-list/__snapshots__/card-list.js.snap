// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> snapshot 1`] = `
<ul
  className="flex flex-col flex-wrap justify-between gap-y-[8px] sm:flex-row sm:gap-x-[16px] sm:gap-y-[16px]"
>
  <li
    className="flex"
  >
    <div>
      0
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      1
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      2
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      3
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      4
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      5
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      6
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      7
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      8
    </div>
  </li>
  <li
    className="flex"
  >
    <div>
      9
    </div>
  </li>
</ul>
`;
